pkg/cgroups/cgroups.go

```diff
@@ -763,3 +764,84 @@ func HostCgroupRoot() (string, error) {
 	)
 	return "", fmt.Errorf("failed to set cgroup root: %w", err)
 }
+
+// CgroupIDFromPID returns the cgroup id for a given pid.
+func CgroupIDFromPID(pid uint32) (uint64, error) {
+	cgroupFile := fmt.Sprintf("%s/%d/cgroup", option.Config.ProcFS, pid)
+	data, err := os.ReadFile(cgroupFile)
+	if err != nil {
+		return 0, err
+	}
+
+	pathPrefix := fmt.Sprintf("%s/1/root/sys/fs/cgroup", option.Config.ProcFS)
+
+	var path string
+	v2Prefix := "0::"
+	v1Prefix := fmt.Sprintf("%d:%s:", GetCgrpSubsystemIdx(), GetCgrpControllerName())
+
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, v2Prefix) {
+			path = fmt.Sprintf("%s/%s", pathPrefix, line[len(v2Prefix):])
+			break
+		}
+		// TODO: test cgroup v1 implementation
+		if strings.HasPrefix(line, v1Prefix) {
+			path = fmt.Sprintf("%s/%s/%s", pathPrefix, GetCgrpControllerName(), line[len(v2Prefix):])
```

Comment:
here you are using v2prefix where it should be v1prefix

Reply:
Here it is a cgroupv1 bug, if fixed I don't see a reason why it should not work if all the logic runs after our cgroup environment discovery

Reply:
Ah, nice catch! Thanks! Fixed.